build failure of tests/subsys/llext with fvp_baser_aemv8r_aarch32_smp

**Describe the bug**
Building the tests/subsys/llext  for the fvp_baser_aemv8r_aarch32_smp  target

` $ west build -p always -b fvp_baser_aemv8r_aarch32_smp tests/subsys/llext`

fails with 
```
$ZEPHYR_BASE/subsys/llext/llext.c: In function 'llext_link':
$ZEPHYR_BASE/subsys/llext/llext.c:784:25: warning: implicit declaration of function 'arch_dcache_flush_range'; did you mean 'arch_icache_flush_range'? [-Wimplicit-function-declaration]
  784 |                         arch_dcache_flush_range(ext->mem[i], ext->mem_size[i]);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~
      |                         arch_icache_flush_range

```

With that target, the #define CONFIG_CACHE_MANAGEMENT 1

This could be a regression coming from commit 173217753098d6b46cf3d510d0bda5c3dd2b51a8  of the PR https://github.com/zephyrproject-rtos/zephyr/pull/66144



**Impact**
CI failure

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows) ubuntu
 - Toolchain (e.g Zephyr SDK, ...)  SDK 0.16.3
 - Commit SHA b6d9ed095d7bc395be602072db2679d201735113

--- a/subsys/llext/Kconfig
+++ b/subsys/llext/Kconfig
@@ -3,6 +3,7 @@
 
 menuconfig LLEXT
 	bool "Linkable loadable extensions"
+	select CACHE_MANAGEMENT if DCACHE
 	help
 	  Enable the linkable loadable extension subsystem
 

--- a/subsys/llext/llext.c
+++ b/subsys/llext/llext.c
@@ -781,7 +781,7 @@ static int llext_link(struct llext_loader *ldr, struct llext *ext, bool do_local
 	/* Make sure changes to ext sections are flushed to RAM */
 	for (i = 0; i < LLEXT_MEM_COUNT; ++i) {
 		if (ext->mem[i]) {
-			arch_dcache_flush_range(ext->mem[i], ext->mem_size[i]);
+			sys_cache_data_flush_range(ext->mem[i], ext->mem_size[i]);
 		}
 	}
 #endif

